File: prac8/fizzbuzz.pl
is_fizz(X) :- 0 is X mod 3.
is_buzz(X) :- 0 is X mod 5.
is_fizzbuzz(X) :- is_fizz(X), is_buzz(X).

do_print(X) :- is_fizzbuzz(X), write('fizz'); is_fizz(X), write('buzz'); is_buzz(X), write('fizzbuzz'); write(X).

fizzbuzz(100) :-
    do_print(100), nl, !.

fizzbuzz(X) :-
    do_print(X), nl,
    Y is X + 1,
    fizzbuzz(Y).

File: assignment/prac1/fizzbuzz.f
c      f77 fizzbuzz.f -o fizzbuzz; ./fizzbuzz
       PROGRAM fizzbuzz
       integer num, ii
       num = 100
       
       do 10 ii = 1, num
              if (MOD(ii, 3) .EQ. 0 .AND. MOD(ii, 5) .EQ. 0) then
                     write (*,*) "fizzbuzz"
              elseif (MOD(ii, 3) .EQ. 0) then
                     write (*,*) "fizz"
              elseif (MOD(ii, 5) .EQ. 0) then
                     write (*,*) "buzz"
              else
                     write (*,*) ii
              endif
10     continue
       STOP
       END
